buffer/src/main/java/io/netty/buffer/NoopResourceLeak.java

```diff
@@ -17,20 +17,24 @@
 
 import io.netty.util.ResourceLeak;
 
-final class NoopResourceLeak implements ResourceLeak {
+import java.util.concurrent.atomic.AtomicBoolean;
 
-    static final NoopResourceLeak INSTANCE = new NoopResourceLeak();
 
-    private NoopResourceLeak() { }
+// Extending AtomicBoolean for intrinsics and saving memory.
+final class NoopResourceLeak extends AtomicBoolean implements ResourceLeak {
+
+    private static final long serialVersionUID = 7874092436796083851L;
 
     @Override
-    public void record() { }
+    public void record() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1186)